purchase_request/models/stock_move.py

```diff
@@ -54,6 +54,12 @@ def _action_cancel(self):
                 except ValueError:
                     activity_type_id = False
                 pr_line = move.created_purchase_request_line_id
+                if pr_line.product_id.responsible_id:
+                    activity_user = pr_line.product_id.responsible_id
+                elif pr_line.request_id.assigned_to:
+                    activity_user = pr_line.request_id.assigned_to
+                else:
```

Comment:
```suggestion
                elif move.picking_id.user_id:
                    activity_user = move.picking_id.user_id
                else:
```
does this makes sense?

Reply:
Well, if there is a responsible in the picking, then it is ok to me that is notified with preference to the user that triggers the action. The preference is then:

* responsible in the product: in order to match Odoo standard notifications for purchases
* responsible of the purchase request
* responsible of the picking
* the user that triggers the action

I just did that.